genie-web/src/main/java/com/netflix/genie/web/agent/resources/AgentFileProtocolResolver.java

```diff
@@ -55,34 +55,57 @@ public AgentFileProtocolResolver(
         this.agentFileStreamService = agentFileStreamService;
     }
 
+    /**
+     * Create a URI for the given remote agent file.
+     *
+     * @param jobId the job id
+     * @param path  the path of the file within the job directory
+     * @return a {@link URI} representing the remote agent file
+     * @throws URISyntaxException if constructing the URI fails
+     */
+    public static URI createUri(final String jobId, final String path) throws URISyntaxException {
+        return new URI(AgentFileProtocolResolver.URI_SCHEME, jobId, path, null);
+    }
+
+    private static String getAgentResourceURIFileJobId(final URI agentUri) {
+        if (!URI_SCHEME.equals(agentUri.getScheme())) {
+            throw new IllegalArgumentException("Not a valid Agent resource URI: " + agentUri);
+        }
+        return agentUri.getHost();
+    }
+
+    private static String getAgentResourceURIFilePath(final URI agentUri) {
+        if (!URI_SCHEME.equals(agentUri.getScheme())) {
+            throw new IllegalArgumentException("Not a valid Agent resource URI: " + agentUri);
+        }
+        return agentUri.getPath();
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public Resource resolve(final String location, final ResourceLoader resourceLoader) {
         log.debug("Attempting to resolve if {} is an Agent file resource or not", location);
         final URI uri;
+        final String jobId;
+        final Path relativePath;
+
         try {
             uri = URI.create(location);
```

Comment:
Why not use `new URI` here so you get the checked exception and catch that instead of `IllegalArgumentException` that `create` wraps the syntax exception in

Reply:
We're implementing an interface here.
So throwing a checked exception would require catching and rewrapping and throwing. i.e. what `create` is already doing.

Beside, this is out of the scope of this PR and i'd rather not touch it.